### PR TITLE
fix: story for `@scalar/api-reference`

### LINKED ISSUE #496: Running Storybook in the api-reference package not working
### Expected Behavior

Expected to open Storybook and be able to view and edit the ApiReference component

### Actual Behavior

Getting errors on each component I try to open 

```TypeError: Failed to fetch dynamically imported module: http://localhost:6006/src/stories/Configure.mdx?import```

```Error: @vitejs/plugin-react can't detect preamble. Something is wrong. See https://github.com/vitejs/vite-plugin-react/pull/11#discussion_r430879201
    at http://localhost:6006/src/stories/Button.jsx:2121:11```

```TypeError: Failed to fetch dynamically imported module: http://localhost:6006/src/stories/Card.stories.ts```

```TypeError: Failed to fetch dynamically imported module: http://localhost:6006/src/stories/CardContent.stories.ts
```
and so on

### Reproduction steps

```bash
1. `cd packages/api-reference`
2. `pnpm storybook`
```

### Version

latest

### What browsers are you seeing the problem on?

_No response_

### PR DESCRIPTION
**Pull request by hanspagel**
This PR fixes the Storybook setup for @scalar/api-reference. It’s starting now, and HMR is working, too. 🎉 

And it adds it (and the other storybook setup for the component library) to the startup script.

Fixes #496 

EDIT: I don’t know how this happened, but I’ve accidentally updated some dependencies?  Let me know if I should split the PR in two.

### CODE REVIEW

File: packages/api-reference/.storybook/preview.css [hunk 1]
```diff
@@ -0,0 +1,61 @@
+body.sb-show-main,
```
Comment (amritk) [inline packages/api-reference/.storybook/preview.css:1]: Nice added the theming!!! Glad to see its getting useage

Comment (amritk) [reply packages/api-reference/.storybook/preview.css:1]: Maybe @cameronrohani can give storybook a look to see if everything is the right color, I just guessed

Comment (amritk) [reply packages/api-reference/.storybook/preview.css:1]: Also maybe we should move this into  @scalar/themes and consume it from components and references.

**Review by amritk — APPROVED** [review]
LGTM. Just one thought of moving the storbook css into themes or something so we can just consume it

**Review by marclave — APPROVED** [review]
looks good to me! we just need to get the build passing then we can merge ❇️

**Pull request merged**